server/athenian/api/internal/logical_repos.py

```diff
@@ -43,3 +47,52 @@ def extract_logical_repo(repo: str, offset: int = 2) -> str:
         return repo.split("/", offset)[offset]
     except IndexError:
         return ""
+
+
+@dataclasses.dataclass(frozen=True)
+class RepoName:
+    """The name of a repository.
+
+    Possible formats:
+
+    - repository name: "owner/reponame"
+    - logical repository name: "owner/reponame/logicalname".
+    - prefixed repository name: "provider.com/owner/reponame"
+    - prefixed logical repository name: "provider.com/owner/reponame/logicalname"
+
+    """
+
+    prefix: Optional[str]
+    organization: str
+    physical: str
+    logical: Optional[str]
+
+    @classmethod
+    def from_prefixed(cls, prefixed_name: str) -> RepoName:
+        """Build the name one of the "prefixed" formats."""
+        if prefixed_name.count("/") < 2:
+            raise ValueError(f"Invalid  prefixed repo name {prefixed_name}")
+
+        prefix, rest = prefixed_name.split("/", 1)
+        org, rest = rest.split("/", 1)
+
+        if "/" in rest:
+            physical, logical = rest.split("/", 1)
+        else:
+            physical = rest
+            logical = None
+        return RepoName(prefix, org, physical, logical)
+
+    @property
+    def is_logical(self) -> bool:
+        """Whether the name refers to a logical repository."""
+        return self.logical is not None
+
+    @property
+    def unprefixed_physical(self) -> str:
```

Comment:
```suggestion
    def physical_name_with_owner(self) -> str:
```
("name_with_owner" is already a term in metadata DB / GraphQL).